pkg/apis/options/authorization.go

```diff
@@ -0,0 +1,53 @@
+package options
+
+import (
+	"github.com/oauth2-proxy/oauth2-proxy/pkg/authorization/engine"
+	"github.com/spf13/pflag"
+)
+
+// Authorization holds configuration options related to trusted requests which
+// would skip authentication
+type Authorization struct {
+	SkipAuthRegex        []string `flag:"skip-auth-regex" cfg:"skip_auth_regex"`
+	SkipAuthRoutes       []string `flag:"skip-auth-route" cfg:"skip_auth_routes"`
+	SkipAuthPreflight    bool     `flag:"skip-auth-preflight" cfg:"skip_auth_preflight"`
+	SkipAuthStripHeaders bool     `flag:"skip-auth-strip-headers" cfg:"skip_auth_strip_headers"`
+	TrustedIPs           []string `flag:"trusted-ip" cfg:"trusted_ips"`
+
+	// internal set after validation
+	rulesEngine *engine.RulesEngine
```

Comment:
Ok cool - I had noticed that trend in validation but wasn't sure where we wanted to build objects from options.

Question: since NewOAuthProxy is quick large and we get code smell warnings for being over 500 lines, is there a place in pkg that is potentially ideal to organize this options/config => object creation that used to happen in validation?

Reply:
I would have the builder in the engine package, so you call `NewRuleEngine(opts.RequestAuthorization)` or something like this at least and then in `NewOAuthProxy` we have to add `requestAuthorization, err := ...` and the dance around that, not much else we can get away with there I don't think.

The problem with that method is that we copy over a million different fields to the struct, these are slowly going to go away as we refactor more and more into smaller chunks. Eventually the `OAuthProxy` will just be the core logic with a bunch of pre-auth and post-auth middlewares, or at least, that's what I'd like to see at least